src/Orchard.Web/Modules/Orchard.Users/Services/UsernameValidationError.cs

```diff
@@ -0,0 +1,33 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Web;
+using Orchard.Localization;
+
+namespace Orchard.Users.Services {
+
+    public enum Severity {
+        Warning,
```

Comment:
There is no such concept in Orchard, please don't add this. A form is either valid or invalid.

Reply:
We made this because if we enable the custom policies some previously created usernames may be not valid anymore so they can't be edite. For example if you set the minimum username length to 10 you can't edit the admin user anymore so by doing this we still allow the backend user to edit and save users and show a warning while frontend users can't register with an invalid username since at frontend the warning are treated as errors.

We could have used two lists, one containing the warnings and one containing the errors but that would have required our username validation method signature to be completely different from the password validation method signature and the handling of the validation results to be different as well. By introducing the severity we can have consistency between the validation methods and the code that handles the validation results is simpler as well.

Reply:
Understood, thanks